chardet/utf1632prober.py

```diff
@@ -0,0 +1,137 @@
+######################## BEGIN LICENSE BLOCK ########################
+# The Original Code is mozilla.org code.
+#
+# The Initial Developer of the Original Code is
+# Netscape Communications Corporation.
+# Portions created by the Initial Developer are Copyright (C) 1998
+# the Initial Developer. All Rights Reserved.
+#
+# Contributor(s):
+#   Jason Zavaglia
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+# 02110-1301  USA
+######################### END LICENSE BLOCK #########################
+from chardet.enums import ProbingState
+from .charsetprober import CharSetProber
+
+
+# This class simply looks for occurrences of zero bytes, and infers
+# whether the file is UTF16 or UTF32 (low-endian or big-endian)
+# For instance, files looking like ( \0 \0 \0 [nonzero] )+
+# appear to be UTF32LE.  Files looking like ( \0 [notzero] )+ appear
+# may be guessed to be UTF16LE.
+
+class UTF1632Prober(CharSetProber):
```

Comment:
Ah, this would make sense as it would be cleaner - you could verify the existence of the correct escape sequences - my implementation is more of a quick and dirty heuristic.  

I feel grokking this state machine would take me days, so if you don't mind I may defer this task if that's ok.